src/Modal/ModalLayer.jsx

```diff
@@ -52,29 +52,31 @@ function ModalLayer({
     };
   }, [isOpen]);
 
-  if (!isOpen) {
+  if (!isOpen && !isHidden) {
```

Comment:
[consideration] It's possible this conditional could remain unchanged given `isOpen` would need to be truthy for `isHidden` to work anyways.